pkg/controller/clone-controller.go

```diff
@@ -167,6 +167,10 @@ func (cc *CloneController) processPvcItem(pvc *v1.PersistentVolumeClaim) error {
 	pvc, err = updatePVC(cc.clientset, pvc, anno, lab)
 	if err != nil {
 		return errors.WithMessage(err, "could not update pvc %q annotation and/or label")
+	} else {
+		if phase == string(v1.PodSucceeded) {
+			cc.deleteClonePods(sourcePod.Namespace, sourcePod.Name, targetPod.Name)
```

Comment:
maybe out of the scope for this, but if either of the pod deletes fails, they will never be retried.